src/Mod/Draft/draftgeoutils/faces.py

```diff
@@ -185,14 +192,25 @@ def create_face(w1, w2):
             faces_fused_list = []
             for faces in faces_list:
                 if len(faces) > 1 :
-                    faces_fused = faces[0].fuse(faces[1:]).removeSplitter().Faces[0]
-                    faces_fused_list.append(faces_fused)
+                    # Below not good if a face is self-intersecting or reversed
+                    #faces_fused = faces[0].fuse(faces[1:]).removeSplitter().Faces[0]
+                    rf = faces[0]
+                    for f in faces[1:]:
+                        rf = rf.fuse(f).removeSplitter().Faces[0]
```

Comment:
It may be possible to call `removeSplitter` only once. That would be more efficient. I would be interested in knowing why fusing multiple faces fails.

Reply:
RemoveSplitter at last action worth a try indeed.

Reply:
Unfortunately not working :

```
                    for f in faces[1:]:
                        rf = rf.fuse(f)  # Not working
                    rf = rf.removeSplitter().Faces[0]
```

![Screenshot from 2025-03-30 09-49-45](https://github.com/user-attachments/assets/954a40ae-0c80-4938-b8b7-9ff13b212dbd)
![Screenshot from 2025-03-30 09-49-35](https://github.com/user-attachments/assets/6dbfd3cd-b8f0-4427-8984-1b912b785a51)